bindings/core/src/method_handler/account.rs

```diff
@@ -0,0 +1,458 @@
+// Copyright 2023 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+#[cfg(feature = "participation")]
+use std::str::FromStr;
+
+use iota_sdk::{
+    client::api::{
+        PreparedTransactionData, PreparedTransactionDataDto, SignedTransactionData, SignedTransactionDataDto,
+    },
+    types::block::{
+        output::{dto::OutputDto, AliasId, NftId, Output, Rent, TokenId},
+        DtoError,
+    },
+    wallet::{
+        account::{
+            types::{AccountBalanceDto, TransactionDto},
+            AccountHandle, AliasOutputOptions, MintTokenTransactionDto, OutputDataDto, OutputOptions,
+            TransactionOptions,
+        },
+        message_interface::AddressWithUnspentOutputsDto,
+        AddressWithAmount, IncreaseNativeTokenSupplyOptions, NativeTokenOptions, NftOptions,
+    },
+};
+use primitive_types::U256;
+
+use super::Result;
+use crate::{method::AccountMethod, Response};
+
+pub(crate) async fn call_account_method(account: &AccountHandle, method: AccountMethod) -> Result<Response> {
+    match method {
+        AccountMethod::BurnNativeToken {
+            token_id,
+            burn_amount,
+            options,
+        } => {
+            let transaction = account
+                .burn_native_token(
+                    TokenId::try_from(&token_id)?,
+                    U256::try_from(&burn_amount).map_err(|_| DtoError::InvalidField("burn_amount"))?,
+                    options.as_ref().map(TransactionOptions::try_from_dto).transpose()?,
+                )
+                .await?;
+            Ok(Response::SentTransaction(TransactionDto::from(&transaction)))
```

Comment:
Can you return the Response and then at the end wrap it in an Ok pls? Very repetitive